### PR TITLE
refactor: align API for selection components

### PR DESCRIPTION
**Pull request by larsrickert**
Relates to #376

Align API of checkbox/radio group, listbox and select:
- Use `SelectOption` type for options property of all components
- Align `modelValue` property of radio and checkbox group to be a primitive value
- Change select `modelValue` property to be a `SelectOption` so a label for the current options can be set (e.g., to display values where no options exist or options are loading but modelValue is still pre-filled)

Example code (note that the same options can now be passed to all components):
```vue
<script lang="ts" setup>
import {
  OnyxCheckboxGroup,
  OnyxListbox,
  OnyxRadioButtonGroup,
  OnyxSelect,
  type SelectOption,
} from "sit-onyx";
import { ref } from "vue";

const options: SelectOption[] = [
  { value: 1, label: "Option 1" },
  { value: 2, label: "Option 2" },
  { value: 3, label: "Option 3" },
];

const simpleValue = ref<number>();
const multipleValue = ref<number[]>([]);
const selectValue = ref<SelectOption>();
</script>

<template>
  <div>
    <OnyxRadioButtonGroup v-model="simpleValue" label="Label" :options="options" />
    <OnyxListbox v-model="simpleValue" label="Label" :options="options" />

    <OnyxCheckboxGroup v-model="multipleValue" label="Label" :options="options" />
    <OnyxSelect v-model="selectValue" label="Label" :options="options" />
  </div>
</template>
```

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: apps/alpha-test-app/src/components/layout-demo/SingleSettingsGroup.vue [hunk 1]
```diff
@@ -1,33 +1,27 @@
...
 const activeSetting = defineModel<Settings>();
 
 const props = defineProps<{
   headline: string;
-  options: SelectionOption[];
+  options: SelectOption[];
   horizontal?: boolean;
 }>();
 
 /** returns the option with the key of the first `true` setting */
```
[the remark this answers is not in the document]
Comment (larsrickert) [reply apps/alpha-test-app/src/components/layout-demo/SingleSettingsGroup.vue:14]: No the if still ensures that the value is true :)
Or do I get your comment wrong?

File: packages/sit-onyx/src/types/forms.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+// - ...
+
+/**
+ * Select option that is e.g. used inside the select, radio button / checkbox group.
+ */
+export type SelectOption<TValue extends SelectOptionValue = SelectOptionValue> = DensityProp & {
+  /**
+   * Value of the option when it is selected.
+   */
+  value: TValue;
```
Comment (JoCa96) [inline packages/sit-onyx/src/types/forms.ts:19]: Please add comment (or rename) that this is the unique identifier for the option in the list

File: packages/sit-onyx/src/types/forms.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  /**
```
Comment (JoCa96) [inline packages/sit-onyx/src/types/forms.ts:20]: Add `data` prop for convenience

File: packages/sit-onyx/src/components/OnyxRadioButtonGroup/OnyxRadioButtonGroup.vue [hunk 4]
```diff
@@ -1,15 +1,12 @@
-<script lang="ts" setup generic="TValue extends SelectionOptionValue">
-import type { TargetEvent } from "../../types/dom";
+<script lang="ts" setup generic="TValue extends SelectOptionValue">
```
Comment (JoCa96) [inline packages/sit-onyx/src/components/OnyxRadioButtonGroup/OnyxRadioButtonGroup.vue:1]: Use SelectOption for modelValue everywhere, as discussed
```suggestion
<script lang="ts" setup generic="TValue extends SelectOption">
```

**Review by JoCa96 — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/types/forms.ts (hunk 3 above)
Comment (larsrickert) [reply packages/sit-onyx/src/types/forms.ts:21]: Lets discuss this with the other devs in a follow up task #951

File: packages/sit-onyx/src/components/OnyxRadioButtonGroup/OnyxRadioButtonGroup.vue (hunk 4 above)
Comment (larsrickert) [reply packages/sit-onyx/src/components/OnyxRadioButtonGroup/OnyxRadioButtonGroup.vue:1]: Lets discuss this with the other devs in a follow up task #951

**Review by JoCa96 — APPROVED** [review]

**Pull request merged**